webapp/src/main/java/io/github/microcks/util/ai/AICopilotHelper.java

```diff
@@ -388,4 +388,100 @@ private static JsonNode followRefIfAny(JsonNode spec, JsonNode referencableNode)
    private static JsonNode getNodeForRef(JsonNode spec, String reference) {
       return spec.at(reference.substring(1));
    }
+
+  /**
+   * Some AI will reuse already present examples and x-microcks-operation in the
+   * spec. This method cleans a specification from its examples.
+   */
+  protected static String removeTagsFromOpenAPISpec(String specification, String operationName) throws Exception {
+    JsonNode specNode;
+    boolean isJson = specification.trim().startsWith("{");
+
+    if (isJson) {
+      specNode = JSON_MAPPER.readTree(specification);
+    } else {
+      specNode = YAML_MAPPER.readTree(specification);
+    }
+
+    // Resolve schemas and Remove examples recursively from the root.
+    resolveReferenceAndRemoveTagsInNode(specNode, specNode);
+    reduceSpecSize(specNode, operationName);
+
+    if (isJson) {
+      return JSON_MAPPER.writerWithDefaultPrettyPrinter().writeValueAsString(specNode);
+    }
+    return YAML_MAPPER.writeValueAsString(specNode);
+  }
+
+  protected static void resolveReferenceAndRemoveTagsInNode(JsonNode specNode, JsonNode node) {
+    JsonNode target = followRefIfAny(specNode, node);
+    if (target.getNodeType() == JsonNodeType.OBJECT) {
+      if (node.has("$ref")) {
+        ((ObjectNode) node).setAll((ObjectNode) target);
+        ((ObjectNode) node).remove("$ref");
+      }
+      if (target.has("examples")) {
+        ((ObjectNode) node).remove("examples");
+      }
+      if (target.has("example")) {
+        ((ObjectNode) node).remove("example");
+      }
+      if (target.has("x-microcks-operation")) {
+        ((ObjectNode) node).remove("x-microcks-operation");
+      }
+      Iterator<Map.Entry<String, JsonNode>> fields = target.fields();
+      while (fields.hasNext()) {
+        resolveReferenceAndRemoveTagsInNode(specNode, fields.next().getValue());
+      }
+    }
+    if (target.getNodeType() == JsonNodeType.ARRAY) {
+      Iterator<JsonNode> elements = target.elements();
+      while (elements.hasNext()) {
+        resolveReferenceAndRemoveTagsInNode(specNode, elements.next());
+      }
+    }
+  }
+
+  protected static void reduceSpecSize(JsonNode specNode, String operationName) throws Exception {
```

Comment:
Wouldn't `filterOpenAPIOperation()` be a more appropriate name?

Reply:
Of course I changed the method name by filterOpenAPISpec()